packages/stark-ui/src/modules/table/components/table.component.ts

```diff
@@ -491,6 +511,19 @@ export class StarkTableComponent extends AbstractStarkUiComponent implements OnI
 
 			return !matchFilter.length || matchFilter.every(Boolean);
 		};
+
+		if (this.filter.globalFilterValue) {
```

Comment:
Perhaps we should improve this check to: `if (this.filter.globalFilterValue && this.filter.globalFilterValue !== "")`. This way we make sure that even an empty string is the same as an empty filter so we avoid triggering the filter in this case (performance gain)